File: shared/contracts/ts/v1/entity-alias.ts
export interface ConfirmEntityAliasRequest {
  alias_text: string;
  canonical_entity_id: string;
  canonical_name: string;
  confidence: number;
}

export interface ConfirmEntityAliasResponse {
  alias_text: string;
  canonical_entity_id: string;
  canonical_name: string;
  confidence: number;
  source: string;
}

export interface EntityAliasCalibrationResponse {
  total_aliases: number;
  avg_confidence: number | null;
}

export interface ResolveEntityInput {
  entity_id: string;
  text: string;
  label: string;
  confidence: number;
}

export interface ResolvedEntityOutput {
  source_entity_id: string;
  source_text: string;
  canonical_entity_id: string;
  canonical_name: string;
  confidence: number;
  matched_layer: string;
}

export interface UnresolvedEntityOutput {
  source_entity_id: string;
  source_text: string;
}

export interface ResolveEntitiesRequest {
  entities: ResolveEntityInput[];
}

export interface ResolveEntitiesResponse {
  resolved: ResolvedEntityOutput[];
  unresolved: UnresolvedEntityOutput[];
}

File: shared/contracts/python/v1/entity_alias.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ConfirmEntityAliasRequest(BaseModel):
    alias_text: str = Field(min_length=1)
    canonical_entity_id: str = Field(min_length=1)
    canonical_name: str = Field(min_length=1)
    confidence: float = Field(ge=0.0, le=1.0)


class ConfirmEntityAliasResponse(BaseModel):
    alias_text: str
    canonical_entity_id: str
    canonical_name: str
    confidence: float
    source: str


class EntityAliasCalibrationResponse(BaseModel):
    total_aliases: int = Field(ge=0)
    avg_confidence: float | None = None


class ResolveEntityInput(BaseModel):
    entity_id: str = Field(min_length=1)
    text: str = Field(min_length=1)
    label: str = Field(min_length=1)
    confidence: float = Field(ge=0.0, le=1.0)


class ResolvedEntityOutput(BaseModel):
    source_entity_id: str
    source_text: str
    canonical_entity_id: str
    canonical_name: str
    confidence: float
    matched_layer: str


class UnresolvedEntityOutput(BaseModel):
    source_entity_id: str
    source_text: str


class ResolveEntitiesRequest(BaseModel):
    entities: list[ResolveEntityInput]


class ResolveEntitiesResponse(BaseModel):
    resolved: list[ResolvedEntityOutput]
    unresolved: list[UnresolvedEntityOutput]
